astropy/cosmology/parameter.py

```diff
@@ -210,8 +212,98 @@ def register(fvalidate):
 
     # -------------------------------------------
 
+    def clone(self, **kw):
+        """Clone this `Parameter`, changing any constructor argument.
+
+        Parameters
+        ----------
+        **kw
+            Passed to constructor. The current values, eg. ``fvalidate`` are
+            used as the default values, so an empty ``**kw`` is an exact copy.
+
+        Examples
+        --------
+        >>> p = Parameter()
+        >>> p
+        Parameter(name=None, derived=False, unit=None, equivalencies=[],
+                  fvalidate='default', fmt='.3g', doc=None)
+
+        >>> p.clone(name="cloned", unit="km")
+        Parameter(name='cloned', derived=False, unit=Unit("km"),
+                  equivalencies=[], fvalidate='default', fmt='.3g', doc=None)
+        """
+        kw.setdefault("doc", self.__doc__)
+        kw.setdefault("fmt", self.format_spec)
+        kw.setdefault("unit", self.unit)
+        kw.setdefault("equivalencies", self.equivalencies)
+        kw.setdefault("derived", self.derived)
+        kw.setdefault("name", self.name)
+        # validator is always turned into a function, but for ``repr`` we want
+        # to know if it's originally a string.
+        fvalidate = (self._validate_repr if self._validate_repr in self._registry_validators
+                     else self.fvalidate)
+        kw.setdefault("fvalidate", fvalidate)
+
+        # all initialization failures, like incorrect input are handled by init
+        return type(self)(**kw)
+
+    def __eq__(self, other):
+        """Check Parameter equality. Only equal to other Parameter objects.
+
+        Returns
+        -------
+        NotImplemented or True
+            `True` if equal, `NotImplemented` otherwise. This allows `other` to
+            be check for equality with ``other.__eq__``.
+
+        Examples
+        --------
+        >>> p1, p2 = Parameter(name="H0"), Parameter(name="H0")
+        >>> p1 == p2
+        True
+
+        >>> p3 = Parameter(name="not H0")
+        >>> p3 == p1
+        False
+
+        >>> p1 != 2
+        True
+        """
+        if not isinstance(other, Parameter):
+            return NotImplemented
+        # check equality on all @property
+        tocheck = (n for n in dir(self.__class__)  # sieve for @property
```

Comment:
Why so fancy when above and below you just use a list of attributes? -- it might make sense to have that list available on the class, a bit like some of the `info` descriptors do (though here there is less expectation of variation).

Reply:
> below you just use a list of attributes

Where below?


> it might make sense to have that list available on the class

SGTM. Marginally faster. It shouldn't change and I don't expect Parameter to be subclassed in a way that changes the properties.